flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -2954,60 +2971,71 @@ struct ZeroOpConversion : public FIROpConversion<fir::ZeroOp> {
   }
 };
 
-// convert to LLVM IR dialect `unreachable`
+/// `fir.unreachable` --> `llvm.unreachable`
 struct UnreachableOpConversion : public FIROpConversion<fir::UnreachableOp> {
   using FIROpConversion::FIROpConversion;
 
   mlir::LogicalResult
-  matchAndRewrite(fir::UnreachableOp unreach, OperandTy operands,
+  matchAndRewrite(fir::UnreachableOp unreach, OpAdaptor adaptor,
                   mlir::ConversionPatternRewriter &rewriter) const override {
     rewriter.replaceOpWithNewOp<mlir::LLVM::UnreachableOp>(unreach);
     return success();
   }
 };
 
-// Check if an argument is present.
+/// `fir.is_present` -->
+/// ```
+///  %0 = llvm.mlir.constant(0 : i64)
+///  %1 = llvm.ptrtoint %0
+///  %2 = llvm.icmp "ne" %1, %0 : i64
+/// ```
 struct IsPresentOpConversion : public FIROpConversion<fir::IsPresentOp> {
   using FIROpConversion::FIROpConversion;
 
   mlir::LogicalResult
-  matchAndRewrite(fir::IsPresentOp isPresent, OperandTy operands,
+  matchAndRewrite(fir::IsPresentOp isPresent, OpAdaptor adaptor,
                   mlir::ConversionPatternRewriter &rewriter) const override {
-    auto idxTy = lowerTy().indexType();
-    auto loc = isPresent.getLoc();
-    auto ptr = operands[0];
+    mlir::Type idxTy = lowerTy().indexType();
+    mlir::Location loc = isPresent.getLoc();
+    auto ptr = adaptor.getOperands()[0];
+
     if (isPresent.val().getType().isa<fir::BoxCharType>()) {
       auto structTy = ptr.getType().cast<mlir::LLVM::LLVMStructType>();
       assert(!structTy.isOpaque() && !structTy.getBody().empty());
-      auto ty = structTy.getBody()[0];
-      auto *ctx = isPresent.getContext();
+
+      mlir::Type ty = structTy.getBody()[0];
+      mlir::MLIRContext *ctx = isPresent.getContext();
       auto c0 = mlir::ArrayAttr::get(ctx, rewriter.getI32IntegerAttr(0));
       ptr = rewriter.create<mlir::LLVM::ExtractValueOp>(loc, ty, ptr, c0);
     }
-    auto c0 = genConstantIndex(isPresent.getLoc(), idxTy, rewriter, 0);
+    mlir::LLVM::ConstantOp c0 =
+        genConstantIndex(isPresent.getLoc(), idxTy, rewriter, 0);
     auto addr = rewriter.create<mlir::LLVM::PtrToIntOp>(loc, idxTy, ptr);
     rewriter.replaceOpWithNewOp<mlir::LLVM::ICmpOp>(
         isPresent, mlir::LLVM::ICmpPredicate::ne, addr, c0);
+
```

Comment:
flang code tends toward being compact and not using blank lines.

Reply:
Sorry about that. I do tend to inject empty lines as that makes code easier for me to read. I'm not sure whether that's my poor vision (I don't get 20/20 even with prescription glasses) or how my brain is wired up.

We can revert this, that's not a problem. But I would really prefer to make such changes on `main` and then sync them back in into `fir-dev` (or perhaps skip that final step altogether). In fact, IMO it would be fine to just submit such an update to `main` as an NFC (without a review). Perhaps once all of this file has been upstreamed?

Basically, I'd like to minimise the traffic between `fir-dev` and `main`.